Flaky test

https://github.com/FerretDB/FerretDB/runs/4726308171

```
--- FAIL: TestListDropDatabase/existing (0.10s)
    handler_test.go:734: Using schema "testlistdropdatabase/existing".
    handler_test.go:801: 
        	Error Trace:	handler_test.go:801
        	Error:      	Not equal: 
        	            	expected: types.Document{m:map[string]interface {}{"databases":(*types.Array)(0xc00018b6e0), "ok":1, "totalSize":29983523, "totalSizeMb":28}, keys:[]string{"databases", "totalSize", "totalSizeMb", "ok"}}
        	            	actual  : types.Document{m:map[string]interface {}{"databases":(*types.Array)(0xc00018b728), "ok":1, "totalSize":29999907, "totalSizeMb":28}, keys:[]string{"databases", "totalSize", "totalSizeMb", "ok"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -3,3 +3,3 @@
        	            	   (string) (len=9) "databases": (*types.Array)({
        	            	-   s: ([]interface {}) (len=3) {
        	            	+   s: ([]interface {}) (len=5) {
        	            	     (types.Document) {
        	            	@@ -39,2 +39,26 @@
        	            	      }
        	            	+    },
        	            	+    (types.Document) {
        	            	+     m: (map[string]interface {}) (len=3) {
        	            	+      (string) (len=5) "empty": (bool) false,
        	            	+      (string) (len=4) "name": (string) (len=10) "testdelete",
        	            	+      (string) (len=10) "sizeOnDisk": (int64) 16384
        	            	+     },
        	            	+     keys: ([]string) (len=3) {
        	            	+      (string) (len=4) "name",
        	            	+      (string) (len=10) "sizeOnDisk",
        	            	+      (string) (len=5) "empty"
        	            	+     }
        	            	+    },
        	            	+    (types.Document) {
        	            	+     m: (map[string]interface {}) (len=3) {
        	            	+      (string) (len=5) "empty": (bool) true,
        	            	+      (string) (len=4) "name": (string) (len=10) "testupdate",
        	            	+      (string) (len=10) "sizeOnDisk": (int64) 0
        	            	+     },
        	            	+     keys: ([]string) (len=3) {
        	            	+      (string) (len=4) "name",
        	            	+      (string) (len=10) "sizeOnDisk",
        	            	+      (string) (len=5) "empty"
        	            	+     }
        	            	     }
        	            	@@ -43,3 +67,3 @@
        	            	   (string) (len=2) "ok": (float64) 1,
        	            	-  (string) (len=9) "totalSize": (int64) 29983523,
        	            	+  (string) (len=9) "totalSize": (int64) 29999907,
        	            	   (string) (len=11) "totalSizeMb": (int64) 28
        	Test:       	TestListDropDatabase/existing
    testutil.go:90: Keeping schema "testlistdropdatabase/existing" for debugging.
logger.go:130: 2022-01-06T11:23:15.953Z	ERROR	pgconn.Pool	Exec	{"err": "ERROR: schema \"nonexisting\" does not exist (SQLSTATE 3F000)", "pid": 108, "sql": "DROP SCHEMA \"nonexisting\" CASCADE", "args": []}
logger.go:130: 2022-01-06T11:23:15.954Z	INFO	pgconn.Pool	closed connection	{"pid": 108}
```

We probably test something in parallel that we should not.

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -8,6 +8,8 @@ run:
 linters-settings:
   exhaustive:
     default-signifies-exhaustive: false
+  goconst:
+    ignore-tests: true
   godot:
     scope: toplevel
     exclude:

--- a/internal/handlers/msg_delete_test.go
+++ b/internal/handlers/msg_delete_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
+// TODO Rework to make them closer to other tests.
 func TestDelete(t *testing.T) {
 	ctx, h, pool := setup(t, nil)
 	schema := testutil.Schema(ctx, t, pool)

--- a/internal/handlers/msg_delete_test.go
+++ b/internal/handlers/msg_delete_test.go
@@ -26,6 +26,7 @@ import (
 )
 
 // TODO Rework to make them closer to other tests.
+//nolint:paralleltest // TODO
 func TestDelete(t *testing.T) {
 	ctx, h, pool := setup(t, nil)
 	schema := testutil.Schema(ctx, t, pool)

--- a/internal/handlers/msg_delete_test.go
+++ b/internal/handlers/msg_delete_test.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package jsonb1
+package handlers
 
 import (
 	"testing"
@@ -21,21 +21,19 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 func TestDelete(t *testing.T) {
-	t.Parallel()
-
-	ctx, h, schema := setup(t)
+	ctx, h, pool := setup(t, nil)
+	schema := testutil.Schema(ctx, t, pool)
 
 	header := wire.MsgHeader{
 		OpCode: wire.OP_MSG,
 	}
 
 	t.Run(schema, func(t *testing.T) {
-		t.Parallel()
-
 		for i := 1; i <= 5; i++ {
 			var msg wire.OpMsg
 			err := msg.SetSections(wire.OpMsgSection{
@@ -138,8 +136,6 @@ func TestDelete(t *testing.T) {
 		for name, tc := range testCases {
 			tc := tc
 			t.Run(name, func(t *testing.T) {
-				t.Parallel()
-
 				tc.req.Set("$db", schema)
 
 				var reqMsg wire.OpMsg

--- a/internal/handlers/msg_update_test.go
+++ b/internal/handlers/msg_update_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
+// TODO Rework to make them closer to other tests.
 func TestUpdate(t *testing.T) {
 	ctx, h, pool := setup(t, nil)
 	schema := testutil.Schema(ctx, t, pool)

--- a/internal/handlers/msg_update_test.go
+++ b/internal/handlers/msg_update_test.go
@@ -26,6 +26,7 @@ import (
 )
 
 // TODO Rework to make them closer to other tests.
+//nolint:paralleltest // TODO
 func TestUpdate(t *testing.T) {
 	ctx, h, pool := setup(t, nil)
 	schema := testutil.Schema(ctx, t, pool)

--- a/internal/handlers/msg_update_test.go
+++ b/internal/handlers/msg_update_test.go
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package jsonb1
+package handlers
 
 import (
 	"strconv"
@@ -21,13 +21,13 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 func TestUpdate(t *testing.T) {
-	t.Parallel()
-
-	ctx, h, schema := setup(t)
+	ctx, h, pool := setup(t, nil)
+	schema := testutil.Schema(ctx, t, pool)
 
 	header := &wire.MsgHeader{
 		OpCode: wire.OP_MSG,

--- a/internal/handlers/sql/convert_test.go
+++ b/internal/handlers/sql/convert_test.go
@@ -22,16 +22,11 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/FerretDB/FerretDB/internal/types"
-	"github.com/FerretDB/FerretDB/internal/util/testutil"
 )
 
 func TestConvert(t *testing.T) {
 	t.Parallel()
-
-	ctx := testutil.Ctx(t)
-	pool := testutil.Pool(ctx, t, &testutil.PoolOpts{
-		ReadOnly: true,
-	})
+	ctx, pool := setup(t)
 
 	rows, err := pool.Query(ctx, "SELECT * FROM pagila.actor ORDER BY actor_id")
 	require.NoError(t, err)

--- a/internal/handlers/sql/sql_test.go
+++ b/internal/handlers/sql/sql_test.go
@@ -12,30 +12,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package jsonb1
+package sql
 
 import (
 	"context"
 	"testing"
 
-	"go.uber.org/zap/zaptest"
-
-	"github.com/FerretDB/FerretDB/internal/handlers"
+	"github.com/FerretDB/FerretDB/internal/pg"
 	"github.com/FerretDB/FerretDB/internal/util/testutil"
 )
 
-func setup(tb testing.TB) (context.Context, *handlers.Handler, string) {
-	ctx := testutil.Ctx(tb)
-	pool := testutil.Pool(ctx, tb, new(testutil.PoolOpts))
-	db := testutil.Schema(ctx, tb, pool)
-	handlerOpts := &handlers.NewOpts{
-		PgPool:        pool,
-		Logger:        zaptest.NewLogger(tb),
-		SharedHandler: nil,
-		SQLStorage:    nil,
-		JSONB1Storage: NewStorage(pool, zaptest.NewLogger(tb)),
-		Metrics:       handlers.NewMetrics(),
-	}
-	h := handlers.New(handlerOpts)
-	return ctx, h, db
+// All tests in this package should be fine running in parallel with tests from the handler package.
+// For that reason, we use read-only access.
+//
+// Exported methods should be tested by the handler package.
+func setup(t *testing.T) (context.Context, *pg.Pool) {
+	ctx := testutil.Ctx(t)
+	pool := testutil.Pool(ctx, t, &testutil.PoolOpts{
+		ReadOnly: true,
+	})
+	return ctx, pool
 }

